kratos/utilities/rbf_shape_functions_utility.cpp

```diff
@@ -21,26 +21,49 @@
 
 namespace Kratos
 {
-
-    double RBFShapeFunctionsUtility::EvaluateRBF(
-        const double x,
-        const double h)
+    template <bool TApplyOnVector, class TOperation>
+    void BuildRBFSystem(Matrix& rMatrixTarget,
+                                  Vector* pVectorTarget,
```

Comment:
```suggestion
                                  Vector& rVectorTarget,
```

pls no raw pointers unless absolutely necessary

Reply:
I got rid of the raw pointers by overloading `BuildRBFSystem()` for the 2 operations we perform in this code. Does it look good to you now?